diffpy/snmf/stretchednmfapp.py

```diff
@@ -0,0 +1,36 @@
+import numpy as np
+import argparse
+
+from diffpy.snmf.io import load_input_signals, initialize_variables
+
+
+def create_parser():
+    parser = argparse.ArgumentParser(
+        prog="stretched_nmf",
+        description="Stretched Nonnegative Matrix Factorization"
+    )
+    parser.add_argument('-v', '--version', action='version', help='Print the software version number')
+    parser.add_argument('-d', '--directory', type=str,
+                        help="Directory containing experimental data. Ensure it is in quotations or apostrophes.")
```

Comment:
by default, the default value of an optional argument is None (so it would be redundant to explicitly set that). The load_input_signal function handles the case where it is None as you described. 

When I was using it, the cli wouldn't work when I specified a directory and it wasn't in quotes. I can see if I can fix this in the code. 

I will rename the variables and add an argument as you described.

Reply:
ok, sounds good.  

The most important thing is that the users know what is happening.  They don't need to know "the default value is None and this is handled in the code" they need to know the arg is optional and the default behavior is that it uses the current working directory.

It is always a good idea to make your intentions clearer for other devs who come later, so I could definitely argue in favor of explicitly defining default to be None.  It just reduces cognitive overload on the person coming later....they don't have to scratch their head and think, "what is going on?" and having to remember what the default behavior of argparse is....